### PR TITLE
Fetch model statuses and store in Redux

### PR DESCRIPTION
**Pull request by hatched**
## Done

Fetch the model status for the list of models returned by modelList and then store that data in the Redux under the model.

## QA

- Pull code
- Run `./run clean && ./run serve`
- Open http://0.0.0.0:8036/
- Click the link to log into the controller and then switch to redux devtools
- You will see the model status values being updated.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -27,12 +27,40 @@ const reduxStore = createStore(
 // eslint-disable-next-line no-shadow
 async function connectAndListModels(reduxStore) {
   try {
+    // eslint-disable-next-line no-console
+    console.log("Logging into the Juju controller.");
```
Comment (hatched) [inline src/index.js]: These console logs will go away once we have the UI updating.

File: src/index.js [hunk 2]
```diff
@@ -27,12 +27,40 @@ const reduxStore = createStore(
...
+    // listModels to update the model lists.
+    const modelList = reduxStore.getState().juju.models.items;
+    // eslint-disable-next-line no-console
+    console.log("Fetching model statuses");
+
+    // Using a for loop here for performance reasons for users with many models.
+    // eslint-disable-next-line no-restricted-syntax
+    for (const modelUUID in modelList) {
+      if (!Object.prototype.hasOwnProperty.call(modelList, modelUUID)) {
+        // eslint-disable-next-line no-continue
```
Comment (hatched) [inline src/index.js]: I'm not sure why `no-continue` is enabled??

File: src/juju/reducers.js [hunk 3]
```diff
@@ -2,21 +2,31 @@ import produce from "immer";
 
 import { actionsList } from "./actions";
 
-export default function jujuReducers(state = {}, action) {
-  return produce(state, draftState => {
+export default produce(
+  (draftState, action) => {
+    // The below is required because CRA ignores the eslint configuration file
+    // during development and adds warnings in the dev console.
+    // eslint-disable-next-line default-case
```
Comment (hatched) [inline src/juju/reducers.js]: I'm open to suggestions on how to avoid those warnings without resorting to weird hacks like this.

Comment (barrymcgee) [reply src/juju/reducers.js]: Add a blank default?

```
      case actionsList.updateModelStatus:
        const payload = action.payload;
        draftState.models.items[payload.modelUUID].status = payload.status;
        break;
      default: // do nothing.
```

Comment (hatched) [reply src/juju/reducers.js]: It was then complaining about an empty default haha.

Comment (hatched) [reply src/juju/reducers.js]: I could then add a console log and then an exception for that 😆

**Comment by hatched** [comment]
Thanks!

**Pull request merged**